### PR TITLE
[C+B] Add a world broadcast and broadcastMessage function

### PR DESCRIPTION
**Pull request by kezz**
This PR adds two new functions to the CraftWorld class, broadcast and broadcastMessage. They exactly mimic CraftServer's broadcast and broadcastMessage except CraftWorld's version will only broadcast the message to players in that world.

I added it because I use lots of plugins that iterate through every single player, check the world they are in and then send them a message. It is a lot easier and less time consuming to use something like `player.getWorld().broadcastMessage("Hello people in my world!");`

CraftBukkit and Bukkit both compile successfully using Maven with this awesome addition.

To demonstrate this I have made a plugin that just has one command

> /test [worldName] [permission]  

If you want to see this in action for yourself you can download the plugin [here](https://www.dropbox.com/s/mgwgxzw6r44b4xv/WBT.jar?dl=1) and the modified CraftBukkit [here](https://www.dropbox.com/s/snb87sdiqxydp73/craftbukkit-1.4.7-R1.1-SNAPSHOT.jar?dl=1).

See https://github.com/Bukkit/CraftBukkit/pull/1033
See https://bukkit.atlassian.net/browse/BUKKIT-3638

### CODE REVIEW

**Comment by turt2live** [comment]
There should be a leaky ticket and a use case applied with this

File: src/main/java/org/bukkit/World.java [hunk 1]
```diff
@@ -1061,6 +1061,23 @@
...
+     * Broadcasts the specified message to every player in this world with the given permission
+     * 
+     * @param message the message 
+     * @param permission the permission
+     * @return the number of players who recieved the message
+     */
+    public int broadcast(String message, String permission);
+    
+    /**
+     * Broadcasts the specified message to every player in this world
```
Comment (riking) [inline src/main/java/org/bukkit/World.java:1076]: Suggestion:

```
Broadcasts the specified message to every player in this world with the {@link org.bukkit.Server#BROADCAST_CHANNEL_USERS} permission
```

**Comment by kezz** [comment]
@turt2live All the information you specified is available on the matching CraftBukkit pull request which I linked to above.

File: src/main/java/org/bukkit/World.java (hunk 1 above)
Comment (kezz) [inline src/main/java/org/bukkit/World.java:1076]: Hmm could do... My only thought is that it it not specified on the normal broadcast... Hmm...

**Comment by riking** [comment]
@kezz101 You generally put all the information on both.
Also, try adding a [C+B] to the issue title

**Comment by kezz** [comment]
@riking All done :+1:

File: src/main/java/org/bukkit/World.java [hunk 2]
```diff
@@ -1061,6 +1061,23 @@
      * @return True if rule exists
      */
     public boolean isGameRule(String rule);
+
```
Comment (natemort) [inline src/main/java/org/bukkit/World.java]: You've got spaces on this line.

File: src/main/java/org/bukkit/World.java [hunk 3]
```diff
@@ -1061,6 +1061,23 @@
...
+    
+    /**
+     * Broadcasts the specified message to every player in this world with the given permission
+     * 
+     * @param message the message 
+     * @param permission the permission
+     * @return the number of players who recieved the message
+     */
+    public int broadcast(String message, String permission);
+
```
Comment (natemort) [inline src/main/java/org/bukkit/World.java]: Spaces.

File: src/main/java/org/bukkit/World.java [hunk 4]
```diff
@@ -1061,6 +1061,23 @@
      * @return True if rule exists
      */
     public boolean isGameRule(String rule);
+    
+    /**
+     * Broadcasts the specified message to every player in this world with the given permission
+     *
```
Comment (gmcferrin) [inline src/main/java/org/bukkit/World.java]: Extra spaces.

File: src/main/java/org/bukkit/World.java [hunk 5]
```diff
@@ -1061,6 +1061,23 @@
...
+     * 
+     * @param message the message 
+     * @param permission the permission
+     * @return the number of players who recieved the message
+     */
+    public int broadcast(String message, String permission);
+    
+    /**
+     * Broadcasts the specified message to every player in this world
+     *
```
Comment (gmcferrin) [inline src/main/java/org/bukkit/World.java]: Extra spaces.

**Comment by kezz** [comment]
I'm going to close this and reopen it as my git messed up.

**Pull request closed without merging**